kitty/fontconfig.c

```diff
@@ -37,7 +37,21 @@ pattern_as_dict(FcPattern *pat) {
         if (PyDict_SetItemString(ans, #name, p) != 0) { Py_CLEAR(p); Py_CLEAR(ans); return NULL; } \
         Py_CLEAR(p); \
     }}
+
+#define L(type, get, which, conv, name) { \
+    type out; PyObject *p; int n = 0; \
+    PyObject *l = PyList_New(0); \
+    while (get(pat, which, n, &out) == FcResultMatch) { \
+        p = conv(out); if (p == NULL) { Py_CLEAR(l); Py_CLEAR(ans); return NULL; } \
+        if (PyList_Append(l, p) != 0) { Py_CLEAR(p); Py_CLEAR(l); Py_CLEAR(ans); return NULL; } \
+        Py_CLEAR(p); \
+        n++; \
+    } \
+    if (PyDict_SetItemString(ans, #name, l) != 0) { Py_CLEAR(l); Py_CLEAR(ans); return NULL; } \
+    Py_CLEAR(l); \
+}
```

Comment:
Please doublecheck that one @kovidgoyal, I tried to make sure I've cleaned up properly but I might have missed something.